tools/mtouch/Target.cs

```diff
@@ -622,19 +622,32 @@ public void ManagedLink ()
 					                               $"The linker output contains more than one assemblies named '{group.Key}':\n\t{string.Join ("\n\t", group.Select ((v) => v.MainModule.FileName).ToArray ())}");
 			}
 
+			Assembly dummy = null;
+			if (App.EnableDedup) {
+				foreach (var a in Assemblies)
+					if (a.IsDedupDummy)
+						dummy = a;
+			}
+
 			// Update (add/remove) list of assemblies in each app, since the linker may have both added and removed assemblies.
 			// The logic for updating assemblies when doing code-sharing is not equivalent to when we're not code sharing
 			// (in particular code sharing is not supported when there are xml linker definitions), so we need
 			// to maintain two paths here.
 			if (sharingTargets.Count == 0) {
 				Assemblies.Update (this, output_assemblies);
+				if (dummy != null)
```

Comment:
What happens if you add `dummy` to `output_assemblies` a few lines above (so that you don't duplicate logic here and below)?

Reply:
That would prevent the duplication here but output_assemblies aren't added to each target in the lower portion